src/editor/codeeditor.cpp

```diff
@@ -740,17 +757,30 @@ void formating(QTextCharFormat fmt, QTextCursor &cursor, Token token, int starti
 void CodeEditor::highlightText()
 {
     // Set cursor position to begining of visible area
-    QTextCursor cursor = this->cursorForPosition(QPoint(0, 0));
-    int firstVisibleLine = cursor.blockNumber();
+//    QTextCursor cursor = this->cursorForPosition(QPoint(0, 0));
+//    int firstVisibleLine = cursor.blockNumber();
+//    int startingPosition = cursor.position();
+
+    QTextBlock block = document()->findBlockByLineNumber(mHighlightingStart);
+    QTextCursor cursor(block);
+    int start = cursor.blockNumber();
     int startingPosition = cursor.position();
 
+//    for(int i = 0; i < mHighlightingStart; ++i)
+//    {
+//        cursor.setPosition(startingPosition);
+//        cursor.movePosition(QTextCursor::EndOfLine);
+//        startingPosition = cursor.position() + 1;
+//    }
+//    qDebug() << startingPosition;
+
     // Set cursor to end of visible aread
     QPoint bottom_right(this->viewport()->width() - 1, this->viewport()->height() - 1);
     cursor = this->cursorForPosition(bottom_right);
     int lastVisibleLine = cursor.blockNumber();
 
     // Highlight visible area
-    for(int i = firstVisibleLine; i <= lastVisibleLine; ++i)
+    for(int i = start; i <= lastVisibleLine; ++i)
```

Comment:
1) for (int i = start; i <= lastVisibleLine; ++i)
2) better use **auto** . _i_ can't be negative, so you don't need to create type as int